File: 700/739 - Daily Temperatures.java
import java.util.Stack;

class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int len = temperatures.length;
        Stack<Integer> stack = new Stack<>();
        int[] ret = new int[len];
        
        for (int i = 0; i < len; ++i) {
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]) {
                int top = stack.pop();
                ret[top] = i - top;
            }
            
            stack.push(i);
        }
        
        return ret;
    }
}

File: csharp/739.daily-temperatures.cs
/*
 * @lc app=leetcode id=739 lang=csharp
 *
 * [739] Daily Temperatures
 */

using System.Collections.Generic;

// @lc code=start
public class Solution {
    public int[] DailyTemperatures(int[] T) {
        IList<int> stack = new List<int>();
        int n = T.Length;
        int[] ret = new int[n];
        for (int i = 0; i < n; ++i) {
            int t = T[i];
            for (int j = stack.Count - 1; j >= 0; --j) {
                int p = stack[j];
                if (T[p] >= t) {
                    break;
                }
                stack.RemoveAt(j);
                ret[p] = i - p;
            }
            stack.Add(i);
        }
        return ret;
    }
}
// @lc code=end

